From 0b3599bb7547d0a1093a49e772fd87a07eb8096c Mon Sep 17 00:00:00 2001
From: Ash Charles <ashcharles@gmail.com>
Date: Wed, 14 Jan 2015 10:51:25 -0800
Subject: [PATCH 14/32] omap3isp: Add missing ioctls

Add V4L2 ioctls for the OMAP3 ISP to report the video format as
UYVY0.

Originally from [1].

[1] https://github.com/gumstix/meta-gumstix/pull/13.patch

Signed-off-by: Ash Charles <ashcharles@gmail.com>
---
 drivers/media/platform/omap3isp/ispvideo.c | 43 ++++++++++++++++++++++++++++++
 1 file changed, 43 insertions(+)

diff --git a/drivers/media/platform/omap3isp/ispvideo.c b/drivers/media/platform/omap3isp/ispvideo.c
index 948fbda797f4..04f6839a2c27 100644
--- a/drivers/media/platform/omap3isp/ispvideo.c
+++ b/drivers/media/platform/omap3isp/ispvideo.c
@@ -1201,6 +1201,47 @@ isp_video_s_input(struct file *file, void *fh, unsigned int input)
 	return input == 0 ? 0 : -EINVAL;
 }
 
+static int
+isp_video_enum_format(struct file *file, void *fh, struct v4l2_fmtdesc *fmtdesc)
+{
+	struct isp_video_fh *vfh = to_isp_video_fh(fh);
+	struct isp_video *video = video_drvdata(file);
+
+	if (fmtdesc->index) {
+		printk(KERN_ERR "%s: invalid index (%d)\n",
+				__func__, fmtdesc->index);
+		return -EINVAL;
+	}
+
+	if (fmtdesc->type != video->type) {
+		printk(KERN_ERR "%s: invalid type (%d). Current type: %d\n",
+				__func__, fmtdesc->type, video->type);
+		return -EINVAL;
+	}
+
+	/* Ugly hack */
+	fmtdesc->flags = 0;
+	fmtdesc->description[0] = 'U';
+	fmtdesc->description[1] = 'Y';
+	fmtdesc->description[2] = 'V';
+	fmtdesc->description[3] = 'Y';
+	fmtdesc->description[4] = '0';
+	fmtdesc->pixelformat = V4L2_PIX_FMT_UYVY;
+
+	return 0;
+}
+
+static int
+isp_video_queryctrl(struct file *file, void *fh, struct v4l2_queryctrl *queryctrl)
+{
+	if (queryctrl->id < V4L2_CID_BASE) {
+		return -EDOM;
+	}
+
+	return -EINVAL;
+}
+
+
 static const struct v4l2_ioctl_ops isp_video_ioctl_ops = {
 	.vidioc_querycap		= isp_video_querycap,
 	.vidioc_g_fmt_vid_cap		= isp_video_get_format,
@@ -1209,6 +1250,8 @@ static const struct v4l2_ioctl_ops isp_video_ioctl_ops = {
 	.vidioc_g_fmt_vid_out		= isp_video_get_format,
 	.vidioc_s_fmt_vid_out		= isp_video_set_format,
 	.vidioc_try_fmt_vid_out		= isp_video_try_format,
+	.vidioc_enum_fmt_vid_cap	= isp_video_enum_format,
+	.vidioc_queryctrl		= isp_video_queryctrl,
 	.vidioc_cropcap			= isp_video_cropcap,
 	.vidioc_g_crop			= isp_video_get_crop,
 	.vidioc_s_crop			= isp_video_set_crop,
-- 
2.1.4

